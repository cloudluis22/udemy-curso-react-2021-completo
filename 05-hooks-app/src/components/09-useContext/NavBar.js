import React from 'react'
import { Link, NavLink } from 'react-router-dom'
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <Link to='/' className="navbar-brand">useContext Hook</Link>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink exact to='/' className="nav-link" aria-current="page" activeClassName="active">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact to='/login' className="nav-link" activeClassName="active">Login</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact to='/about' className="nav-link" activeClassName="active">About</NavLink>
                    </li>
                </ul>
                </div>
            </div>
        </nav>
    )
}
