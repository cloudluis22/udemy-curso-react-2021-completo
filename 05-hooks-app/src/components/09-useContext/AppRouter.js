import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from 'react-router-dom'

import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>

            <NavBar />

            <div>

                <Switch>

                    <Route exact path="/" component={HomeScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />

                    {/* Si no encuentra una ruta espec√≠fica, redirige a la ruta por defecto. */}
                    <Redirect to='/' />

                </Switch>

            </div>

        </Router>
    )
}
