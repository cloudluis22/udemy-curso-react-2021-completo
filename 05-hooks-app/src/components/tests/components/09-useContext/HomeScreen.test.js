import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from '../../../09-useContext/HomeScreen';
import { UserContext } from '../../../09-useContext/UserContext';

describe('Pruebas en el componente funciona de "<HomeScreen />".', () => {

    const user = {
        user: 'cloudluis22',
        email: 'cloudluis22@hotmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
    )




    test('Debe de mostrarse correctamente y generar una snapshot.', () => {

        expect(wrapper).toMatchSnapshot();

    });

});