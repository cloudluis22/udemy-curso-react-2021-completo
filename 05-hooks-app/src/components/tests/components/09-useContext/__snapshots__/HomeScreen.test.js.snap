// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente funciona de "<HomeScreen />". Debe de mostrarse correctamente y generar una snapshot. 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
       "user": "cloudluis22",
       "email": "cloudluis22@hotmail.com"
    }
  </pre>
</div>
`;
